ql/src/java/org/apache/hadoop/hive/ql/io/parquet/timestamp/NanoTimeUtils.java

```diff
@@ -102,10 +106,22 @@ public static Timestamp getTimestamp(NanoTime nt, ZoneId targetZone, boolean leg
 
     JulianDate jDateTime;
     jDateTime = JulianDate.of((double) julianDay);
+    LocalDateTime localDateTime;
```

Comment:
can we use `localDateTime.isLeapYear()` check?

Reply:
Not exactly, we don't need to have this check for every leap year. 

 The Julian calendar defines a leap year as once every four years. The Gregorian calendar modified the addition of leap days, such that a century year was only counted as a leap year if it was also divisible by 400.
 
So according to julian calender the year 200 or 300 is a leap year but they are not a leap year according to according to Gregorian calendar. That is why we are seeing this issue, this is an edge case.

Reply:
relying on DateTimeException exception is not a good practice, what if the error message changes?

Reply:
Agreed , I have replaced it with a conditional check .